File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    const numbers = [];

    const _dfs = (node, numStr = "") => {
        if (!node) return;
        numStr += node.val.toString();

        if (!node.left && !node.right){
            numbers.push(parseInt(numStr))
        }

        _dfs(node.left, numStr);
        _dfs(node.right, numStr);
    }

    _dfs(root);

    let res = 0;
    for (const num of numbers) {
        res += num;
    }

    return res;
};

File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        numbers = []

        def _dfs(node, numStr = ""):
            if not node: return
            numStr += f"{node.val}"
            
            if not node.left and not node.right:
                numbers.append(numStr)

            _dfs(node.left, numStr)
            _dfs(node.right, numStr)

        _dfs(root)
        
        res = 0

        for num in numbers:
            res += int(num)

        return res